CounterpointGenerator/CounterpointGenerator/Generator.cs

```diff
@@ -40,21 +40,36 @@ private List<MelodyLine> GenerateCounterpointForNoteStack(MelodyLine m, Note? pr
 
         }
 
-        private List<Note> CounterpointForNote(Note n, Note prevN, Note preCtp)
+        private List<Note> CounterpointForNote(Note n, Note prevN, Note preCtp, int melodyLength, int generateCount)
         {
             List<Note> possibleNotes = GenerateNoteAtRegion(n);
-            List<Note> possibleNotesAfterRules = new List<Note>();
-            /*foreach(IRules r in Dictionary[input.userPreference])
+            // Possibilities, currentNote, position, length, previousCantus, previousCounterpoint
+            RuleInput ruleInput = new RuleInput(possibleNotes, n, generateCount, melodyLength, prevN, preCtp);
+
+            ruleApplier = new RuleApplier();
+            if (ruleApplier.GenerateSet())
+            {
+                ruleApplier.Applicator(ruleInput);
+            }
+            else
             {
-                List<Note> newPossibleNotes = r.Apply(possibleNotes);
-                possibleNotesAfterRules.AddRange(newPossibleNotes);
-            }*/
-            return possibleNotesAfterRules;
+                throw new Exception("Ruleset did not generate properly!");
+            }
+
+            return ruleInput.Possibilities;
         }
 
         private List<Note> GenerateNoteAtRegion(Note n)
         {
-            throw new NotImplementedException();
+            int upperPitch = n.Pitch + 13; // +1 octave
+            int lowerPitch = n.Pitch - 13; // -1 octave
+            List<Note> output = new List<Note>();
+            for (int addPitch = lowerPitch; addPitch <= upperPitch; addPitch++)
+            {
+                // TODO: Fix for note length when that becomes something to worry about
```

Comment:
List<note> output = Enumerable.Range(n.Pitch - 13, n.Pitch + 13).Select(pitch=>new Note(pitch)).ToList();

Reply:
Oh, that's convenient

Reply:
It would actually need to be 
```csharp
List<Note> test = Enumerable.Range(-13, 27).Select(pitch => new Note(pitch)).ToList()
```
To get the full 13 semitones on either side of the pitch since it's 
```csharp
Enumerable.Range(int start, int count)
```
"Count" not "End"
So the logic would have to be something like (n.Pitch - 13, n,Pitch + (13*2 + 1)) which while the Enumerable trick is cool I honestly think makes it look so much messier it outweighs the benefits of it.

Reply:
Oh, nice catch!
This comes down to personal preference but I think once you have constants in place (from one of the other comments) Range(n.pitch-octave, generationRange) is tame enough

Reply:
I, personally, strongly prefer the for loop. It's just super easy to look at and see exactly what it does